tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,284 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+					labelSelectors, orgID, preRuleNameListMap[namespace], preRuleUid, postRuleNameListMap[namespace], postRuleUid, periodicPolicyName, schPolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of creating schedule backup with schedule name - %s", scheduleName))
+				schBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup - %s", schBackupName))
+				backupNames = append(backupNames, schBackupName)
+				scheduleNames = append(scheduleNames, scheduleName)
+				backupNamesMap[namespace] = backupNames
+				threadCount = threadCount + len(backupNames)
+			}
+			log.Infof("The Map of backups taken are: %v", backupNamesMap)
+		})
+
+		var sem = make(chan struct{}, threadCount)
+		var wg sync.WaitGroup
+		Step("Restoring the backed up applications", func() {
+			log.InfoD("Restoring the backed up applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				namespaceMapping := make(map[string]string)
+				for _, backupName := range backupNamesMap[namespace] {
+					sem <- struct{}{}
+					restoreName = fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+					restoreNames = append(restoreNames, restoreName)
+					time.Sleep(1 * time.Second)
+					namespaceMapping[namespace] = fmt.Sprintf("new-namespace-%v", time.Now().Unix())
+					wg.Add(1)
+					go func(restoreName string, backupName string, namespaceMapping map[string]string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+					}(restoreName, backupName, namespaceMapping)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of restores taken are: %v", restoreNames)
+			log.InfoD("Sleeping for 20 seconds so that the request reaches stork and the restore process is started")
+			time.Sleep(20 * time.Second)
+		})
+
+		Step("Cancelling the ongoing restores", func() {
+			log.InfoD("Cancelling the ongoing restores")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, restoreName := range restoreNames {
+				sem <- struct{}{}
+				wg.Add(1)
+				go func(restoreName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					defer func() { <-sem }()
+					err := DeleteRestore(restoreName, orgID, ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting restore %s while restore is in progress", restoreName))
+				}(restoreName)
+			}
+			wg.Wait()
+			log.InfoD("Sleeping for 30 seconds for the restore cancellation to take place")
+			time.Sleep(30 * time.Second)
```

Comment:
Similar to https://github.com/portworx/torpedo/pull/1267/files#r1151443361, would it or would it not be better to check status (to verify it is done, before doing `wg.Done()`), than to just sleep. Of course, we mostly use postgres or mysql, so we should be fine with "hardcoded" values.
Though, if you check status and wait for it to finish, you won't need to verify in the step below.

Reply:
handled above.